functests/vrf/vrf.go

```diff
@@ -0,0 +1,338 @@
+	})
+})
+
+func validateVRFConnectivity(cs *apiclient.ClientSet, client string, vrfName string, DestIPAddr string) error {
+	pingCommand := []string{"ping", "-I", vrfName, "-c5", DestIPAddr}
+	pod, err := cs.Pods(TestNamespace).Get(context.Background(), client, metav1.GetOptions{})
+	pingStatus, err := pods.ExecCommand(cs, *pod, pingCommand)
+	if err != nil {
+		return err
+	}
+	if strings.Contains(pingStatus.String(), " 0% packet loss") {
+		return nil
+	}
+	return fmt.Errorf("Connectivity test error")
+}
+
+func addVrfNad(NadName string, ifName string, vrfName string) netattdefv1.NetworkAttachmentDefinition {
+	return netattdefv1.NetworkAttachmentDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: NadName,
+			Namespace:    TestNamespace,
+		},
+		Spec: netattdefv1.NetworkAttachmentDefinitionSpec{
+			Config: fmt.Sprintf(`{"cniVersion": "0.4.0", "name": "macvlan-vrf", "plugins": [{"type": "macvlan","master": "%s","ipam": {"type": "static"}},{"type": "vrf","vrfname": "%s"}]}`, ifName, vrfName),
+		},
+	}
+}
+
+func definePodwithVrfStaticIP(name string, namespace string, nodeName string, networks string) *k8sv1.Pod {
+	return &k8sv1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: name,
+			Namespace:    namespace,
+			Annotations:  map[string]string{"k8s.v1.cni.cncf.io/networks": networks},
+		},
+		Spec: k8sv1.PodSpec{
+			RestartPolicy: k8sv1.RestartPolicyNever,
+			Containers: []k8sv1.Container{
+				{
+					Name:    name,
+					Image:   images.For(images.TestUtils),
+					Command: []string{"sleep"},
+					Args:    []string{"INF"},
+				},
+			},
+			NodeSelector: map[string]string{
+				hostnameLabel: nodeName,
+			},
+		},
+	}
+}
+
+func defineHostNetworkdPod(name string, namespace string, nodeName string, hostNetwork bool) *k8sv1.Pod {
+	return &k8sv1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: name,
+			Namespace:    namespace,
+		},
+		Spec: k8sv1.PodSpec{
+			RestartPolicy: k8sv1.RestartPolicyNever,
+			Containers: []k8sv1.Container{
+				{
+					Name:    name,
+					Image:   images.For(images.TestUtils),
+					Command: []string{"sleep"},
+					Args:    []string{"INF"},
+				},
+			},
+			HostNetwork: hostNetwork,
+			NodeSelector: map[string]string{
+				hostnameLabel: nodeName,
+			},
+		},
+	}
+
+}
+
+func getMainNodeInterface(cs *apiclient.ClientSet, nodeName string) string {
+	tempPodDefinition := defineHostNetworkdPod("temppod", TestNamespace, nodeName, true)
+	err := cs.Create(context.Background(), tempPodDefinition)
+	Expect(err).ToNot(HaveOccurred())
+	Eventually(func() k8sv1.PodPhase {
+		tempPod, _ := cs.Pods(TestNamespace).Get(context.Background(), tempPodDefinition.Name, metav1.GetOptions{})
+		return tempPod.Status.Phase
+	}, podWaitingTime, time.Second).Should(Equal(k8sv1.PodRunning))
+	Expect(err).ToNot(HaveOccurred())
+	defIntName, err := pods.DetectDefaultRouteInterface(cs, *tempPodDefinition)
+	Expect(err).ToNot(HaveOccurred())
+	err = namespaces.Clean(TestNamespace, "temppod", cs)
```

Comment:
This cleans also any sriov resources. Sriov may not even be installed.
Consider splitting the function that cleans pods.

Reply:
removed sriov related steps and renamed function